tests/ui/string_as_str.stderr

```diff
@@ -0,0 +1,22 @@
+error: used `as_str` for reference
+  --> $DIR/string_as_str.rs:8:13
+   |
+LL |     let s = my_string.as_str();
+   |             ^^^^^^^^^^^^^^^^^^ help: replace it with: `&my_string`
+   |
+   = note: `-D clippy::string-as-str` implied by `-D warnings`
+
+error: used `as_str` for reference
+  --> $DIR/string_as_str.rs:9:17
+   |
+LL |     fn_with_str(my_string.as_str());
+   |                 ^^^^^^^^^^^^^^^^^^ help: replace it with: `&my_string`
+
+error: used `as_str` for reference
+  --> $DIR/string_as_str.rs:30:13
+   |
+LL |     let a = s.as_str().chars().map(|c| c.is_digit(2)).collect::<Vec<_>>();
+   |             ^^^^^^^^^^ help: replace it with: `&s`
```

Comment:
This suggestion won't work. It should suggest `(&s)`. You can use `clippy_utils::Sugg` with `maybe_paren`  to automatically produce the parens.

Reply:
I don't think we should lint where parenthesis are needed. This feels like the canonical use of `as_str`

Reply:
Avoid to lint this